src/Compilers/Core/Portable/Operations/UnaryOperatorKind.cs

```diff
@@ -40,7 +40,12 @@ public enum UnaryOperatorKind
         /// <summary>
         /// Represents the C# 'false' operator and VB 'IsFalse' operator.
         /// </summary>
-        False = 0x6
+        False = 0x6,
+
+        /// <summary>
+        /// Represents the C# '^' operator.
+        /// </summary>
+        Index = 0x7,
```

Comment:
>Index = 0x7 [](start = 8, length = 11)

Should we have an IIndexOperation instead?

Reply:
I don't think so. On its own, the `^` is a true unary prefix operator (and I think we should treat is as such.

---
In reply to: [185130825](https://github.com/dotnet/roslyn/pull/26405#discussion_r185130825) [](ancestors = 185130825)

Reply:
> On its own, the ^ is a true unary prefix operator (and I think we should treat is as such.

Syntactically, it appears as an unary operator, but semantically it is very different to other unary operators. For example, there is no relation between its operand and the type which provides the Create... method. I think it should be a separate node.


---
In reply to: [185156042](https://github.com/dotnet/roslyn/pull/26405#discussion_r185156042) [](ancestors = 185156042,185130825)